src/Query/Builder.php

```diff
@@ -538,9 +540,26 @@ public function generateCacheKey()
         return md5(serialize(array_values($key)));
     }
 
-    /** @inheritdoc */
-    public function aggregate($function, $columns = [])
+    /** @return ($function is null ? AggregationBuilder : mixed) */
+    public function aggregate($function = null, $columns = ['*'])
     {
+        if ($function === null) {
```

Comment:
This whole block could use some tests (obviously the missing dependency can't be tested) to protect against regressions.

Reply:
Tests added.